src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -825,49 +825,39 @@ private TypedConstant GetDefaultValueArgument(ParameterSymbol parameter, Attribu
         {
             Debug.Assert(argsConsumedCount <= argumentsCount);
 
-            int paramArrayArgCount = argumentsCount - argsConsumedCount;
-            if (paramArrayArgCount == 0)
-            {
-                return new TypedConstant(parameter.Type, ImmutableArray<TypedConstant>.Empty);
-            }
-
-            int argIndex = -1;
             // If there's a named argument, we'll use that
             if (!constructorArgumentNamesOpt.IsDefault)
             {
-                argIndex = constructorArgumentNamesOpt.IndexOf(parameter.Name);
+                int argIndex = constructorArgumentNamesOpt.IndexOf(parameter.Name);
+                if (TryGetNormalParamValue(parameter, constructorArgsArray, argIndex, conversions, out var namedValue))
+                {
+                    return namedValue;
+                }
+
+                // A named argument for a params parameter must be trailing, so expanded params must have a single value
+                var singleValue = new TypedConstant[1] { constructorArgsArray[argIndex] };
+                return new TypedConstant(parameter.Type, singleValue.AsImmutableOrNull());
```

Comment:
Avoid the extra array with `var singleValue = ImmutableArray.Create(constructorArgsArray[argIndex]));`